src/vs/base/browser/indexedDB.ts

```diff
@@ -118,70 +119,52 @@ export class IndexedDB {
 		this.pendingTransactions.push(transaction);
-		// MEMBRANE: List of keys we save using our `/settings` API endpoint
-		// indexeddb name
-		const dbName = this.name;
-		const MEMBRANE_KEYS = [
-			// 'memento/webviewView.membrane.logs',
-			// 'memento/webviewView.membrane.navigator',
-			// 'memento/webviewView.membrane.packages',
-			'/User/settings.json'
-		];
-		// MEMBRANE: Proxy
+
+		const requests: Array<{ prop: string; key: string; value?: any }> = [];
 		const storeProxy = new Proxy(transaction.objectStore(store), {
-			get(target: any, prop: any): any {
-				if (typeof target[prop] === 'function') {
-					return (...args: any[]) => {
-						const result = (target[prop] as Function).apply(target, args);
-
-						if (isUserDataStore(store) || isWorkspaceDb(dbName)) {
-							const isGet = prop === 'get';
-							const isPut = prop === 'put';
-							let key: any;
-							let value: any;
-
-							if (isUserDataStore(store)) {
-								key = args[1];
-								value = args[0];
-							} else if (isWorkspaceDb(dbName)) {
-								value = args[0];
-								key = args[1];
-							}
-
-							if (isGet && MEMBRANE_KEYS.includes(key)) { }
-
-							if (isPut && MEMBRANE_KEYS.includes(key)) {
-								return new Promise((resolve, reject) => {
-									const decoder = new TextDecoder();
-									value = decoder.decode(value);
-									membraneApi('POST', '/settings', JSON.stringify({ key, value }))
-										.then(() => {
-											result.onsuccess = () => {
-												resolve(result.result);
-											};
-										})
-										.catch(reject);
-								});
-							}
-						}
-						return result;
-					};
-				}
-				return target[prop];
-			}
+			get(target: IDBObjectStore, prop: string): any {
+				return (...args: any[]) => {
+					const result = (target[prop as keyof IDBObjectStore] as Function).apply(target, args);
+					requests.push({
+						prop,
+						key: prop === 'get' ? args[0] : args[1],
+						value: prop === 'put' ? args[0] : undefined,
+					});
+					return result;
+				};
+			},
 		});
 
-		const request: IDBRequest<T> | IDBRequest<T>[] = dbRequestFn(storeProxy);
-		return new Promise<T | T[]>((c, e) => {
-			transaction.oncomplete = () => {
-				if (Array.isArray(request)) {
-					c(request.map(r => r.result));
-				} else {
-					c(request.result);
+
+		const originalRequest: IDBRequest<T> | IDBRequest<T>[] = dbRequestFn(storeProxy);
+
+		return new Promise<T | T[]>((resolve, reject) => {
+			transaction.oncomplete = async () => {
+				try {
+					const membraneResults = await handleMembraneRequests(
+						requests.filter(req => isMembraneKey(req.key))
+					);
+
+					if (Array.isArray(originalRequest)) {
+						resolve(originalRequest.map((r, index) => {
+							const membraneReq = requests[index];
+							return isMembraneKey(membraneReq.key)
+								? membraneResults[membraneResults.findIndex(mr => mr.key === membraneReq.key)]
+								: r.result;
```

Comment:
There's a problem here if the same key is read and written at the same time. Instead, I would store the `IDBRequest` in `requests` (like we [wrote down here](https://github.com/membrane-io/vscode/pull/23#discussion_r1767249319)) and find it using `requests.find((r2) => r2.request === r)`.

Side note, instead of writing this:

```
membraneResults[membraneResults.findIndex(mr => mr.key === membraneReq.key)]
```

Why not do this?
```
membraneResults.find(mr => mr.key === membraneReq.key)
```